src/devtools/client/debugger/src/components/Editor/Preview/index.tsx

```diff
@@ -34,40 +52,54 @@ class Preview extends PureComponent {
     codeMirrorWrapper.removeEventListener("mousedown", this.onMouseDown);
   }
 
-  updateListeners(prevProps) {
+  updateListeners() {
     const { codeMirror } = this.props.editor;
     const codeMirrorWrapper = codeMirror.getWrapperElement();
     codeMirror.on("tokenenter", this.onTokenEnter);
+    codeMirror.on("tokenleave", this.onTokenLeave);
     codeMirror.on("scroll", this.onScroll);
     codeMirrorWrapper.addEventListener("mouseup", this.onMouseUp);
     codeMirrorWrapper.addEventListener("mousedown", this.onMouseDown);
   }
 
-  onTokenEnter = ({ target, tokenPos }) => {
+  onTokenEnter = ({ target, tokenPos }: { target: HTMLElement; tokenPos: SourceLocation }) => {
+    const { cx } = this.props;
+
+    if (cx?.isPaused && !this.state.selecting) {
+      this.startPreview(target, tokenPos);
+    }
+  };
+
+  startPreview = debounce((target: HTMLElement, tokenPos: SourceLocation) => {
     const { cx, editor, updatePreview } = this.props;
 
-    if (cx.isPaused && !this.state.selecting) {
+    // Double-check status after timer runs
+    if (cx?.isPaused && !this.state.selecting) {
       updatePreview(cx, target, tokenPos, editor.codeMirror);
     }
+  }, 300);
```

Comment:
I think this is too high. I don't want it to trigger for symbols that the user's cursor is *literally* just passing over, but I also think that it should feel as instant as possible. I would try lowering this to like 50 or 100.